### PR TITLE
Moving up the logic for skip upgrade

### PR DESCRIPTION
**Pull request by kuvinodms**

### CODE REVIEW

**Comment by kuvinodms** [comment]
In continuation with #7347

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1 [hunk 1]
```diff
@@ -146,6 +146,32 @@ function Publish-UpgradedServiceFabricApplication
...
 
+    $ApplicationManifestPath = "$AppPkgPathToUse\ApplicationManifest.xml"
+
+    $names = Get-NamesFromApplicationManifest -ApplicationManifestPath $ApplicationManifestPath
+    if (!$names)
+    {
+        return
+    }
+
+    # If ApplicationName is not specified on command line get application name from Application parameter file.
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1]: Is there a particular reason why you changed order to first get application names from manifest file and then get application name from parameter file?

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1 [hunk 2]
```diff
@@ -146,6 +146,32 @@ function Publish-UpgradedServiceFabricApplication
...
+        $ApplicationName = Get-ApplicationNameFromApplicationParameterFile $ApplicationParameterFilePath
+    }
+
+    if (!$ApplicationName)
+    {
+        Write-Error (Get-VstsLocString -Key EmptyApplicationName)
+    }
+
+    $oldApplication = Get-ServiceFabricApplication -ApplicationName $ApplicationName
+    if ($SkipUpgradeSameTypeAndVersion -And $oldApplication.ApplicationTypeVersion -eq $names.ApplicationTypeVersion)
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1]: check for $oldApplication not being null

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1 [hunk 3]
```diff
@@ -172,29 +196,10 @@ function Publish-UpgradedServiceFabricApplication
...
-        return
-    }
-
     $ApplicationTypeAlreadyRegistered = $false
     if ($Action.Equals('RegisterAndUpgrade') -or $Action.Equals('Register'))
     {
         ## Check existence of the application
-        $oldApplication = Get-ServiceFabricApplication -ApplicationName $ApplicationName
-
         if (!$oldApplication)
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1]: Infact move this check to before checking for skip upgrade

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1 (hunk 1 above)
Comment (kuvinodms) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1]: Both are different. I think order doesn't matter here.
Still ApplicationName variable is being used in fetching old application so placed this right before of it.

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1 [hunk 4]
```diff
@@ -172,49 +203,14 @@ function Publish-UpgradedServiceFabricApplication
...
-    }
-
     $ApplicationTypeAlreadyRegistered = $false
     if ($Action.Equals('RegisterAndUpgrade') -or $Action.Equals('Register'))
     {
-        ## Check existence of the application
-        $oldApplication = Get-ServiceFabricApplication -ApplicationName $ApplicationName
-
-        if (!$oldApplication)
+        if ($oldApplication.ApplicationTypeName -ne $names.ApplicationTypeName)
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1]: even this if check can be moved up

**Pull request merged**